packages/presentation/lib/ui/movie_page/bloc/movie_bloc.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:domain/use_case/request_use_case_coming.dart';
+import 'package:domain/use_case/request_use_case_trending.dart';
+import 'package:presentation/base/bloc.dart';
+import 'package:presentation/navigation/base_arguments.dart';
+import 'package:presentation/ui/movie_details/details_arguments/movie_details_arguments.dart';
+import 'package:presentation/ui/movie_details/movie_details_widget.dart';
+import 'package:presentation/ui/movie_page/bloc/movie_list_tile.dart';
+import 'package:presentation/ui/movie_page/model/movie_row_data.dart';
+
+abstract class MovieBloc extends Bloc<BaseArguments, MovieListTile> {
+  factory MovieBloc(
+    RequestUseCaseComing requestUseCaseComing,
+    RequestUseCaseTrending requestUseCaseTrending,
+  ) =>
+      MovieBlocImpl(
+        requestUseCaseComing,
+        requestUseCaseTrending,
+      );
+
+  void onMovieTap({
+    required MovieRowData movie,
+  });
+}
+
+class MovieBlocImpl extends BlocImpl<BaseArguments, MovieListTile>
+    implements MovieBloc {
+  final _tile = MovieListTile.init();
+  final RequestUseCaseComing _requestUseCaseComing;
+  final RequestUseCaseTrending _requestUseCaseTrending;
+
+  MovieBlocImpl(
+    this._requestUseCaseComing,
+    this._requestUseCaseTrending,
+  );
+
+  @override
+  void initState() async {
+    super.initState();
+    await loadMovie();
+  }
+
+  Future<void> loadMovie() async {
+    final movieResponseTrending = await _requestUseCaseTrending();
+    final movieResponseComing = await _requestUseCaseComing();
+    final movieMapTrending = movieResponseTrending
+        .map(
+          (e) => MovieRowData.fromMovieListResponse(e.movie),
+        )
+        .toList();
+    final movieMapComing = movieResponseComing
+        .map(
+          (e) => MovieRowData.fromMovieListResponse(e.movie),
+        )
+        .toList();
+    handleData(
+      tile: _tile.copyWith(
+        movieTrending: movieMapTrending,
+        movieComing: movieMapComing,
+      ),
```

Comment:
save tike into _tile before passing into handleData

Reply:
u should also update your tile in the class. why?

let's assume the situation that to display the screen we need the data from 2 api calls - A and B

1) tile has empty fields for A and B
2) we load the data from A and update the screen (tile in the bloc is still empty)
3) we load the data from B (tile in the bloc is still empty)

imho, it's not correct